### ISSUE TITLE
Support for Mastodon and Sharkey

### ISSUE THREAD

**Issue by DraconicNEO** [ismailkarsli/lemmy-federate#19]
Was wondering if it could be possible to implement support for this service to be able to work on Mastodon and Sharkey (misskey fork). Now you're probably asking "Draconic, Mastodon and Sharkey don't have Communties" and you'd be right, but they are able to see the groups from software that does support the protocol, and since some Mastodon users do use them to post to Lemmy or mbin it's probably a good idea to have support for them, especially since Lemmy content has very poor reach on Mastodon due to a low amount of followers there. That could be helped if Lemmy federate supported the larger Fediverse microblogging platforms.

**Comment by ismailkarsli** [comment]
TBH I’m not familiar with how Mastodon works. So we’re gonna follow Lemmy communities from Mastodon but not other way around?

**Comment by ismailkarsli** [comment]
I checked their API. They don't provide local subscriber counts in persons (just like Mbin). They also don't have first party group (community) support.

So I don't think it's worth the trouble.

**Comment by DraconicNEO** [comment]
> TBH I’m not familiar with how Mastodon works. So we’re gonna follow Lemmy communities from Mastodon but not other way around?

Yeah that's how I was thinking of doing it, I guess it could federate Mastodon's group functions out to Lemmy and Mbin whenever they get the Feature natively and officially but for now I was thinking of just pulling from other servers since it doesn't have them natively.


> I checked their API. They don't provide local subscriber counts in persons (just like Mbin). They also don't have first party group (community) support.
> 
> So I don't think it's worth the trouble.

Oh I see, I guess that makes sense. It's a lot of work and it wouldn't make sense to put it in when Mastodon itself doesn't support groups right now. I'll close this now but if Mastodon does get group support in the future do you think I should re-open this issue?

**Comment by ismailkarsli** [comment]
> I'll close this now but if Mastodon does get group support in the future do you think I should re-open this issue?

Of course.
